addons/web/static/lib/py.js/lib/py_extras.js

```diff
@@ -112,6 +112,25 @@ function ymd2ord(year, month, day) {
            day;
 }
 
+function get_quarter_number(month) {
+    return Math.ceil(month / 3);
+}
+
+function get_quarter(year, month) {
+    var quarter_number = get_quarter_number(month);
+    var month_from = ((quarter_number - 1) * 3) + 1
+    var date_from = {year: year, month: month_from, day: 1}
+    var date_to = {year: year, month: month_from + 2, day: days_in_month(year, month)}
+    return [date_from, date_to];
+}
+
+function get_day_of_week(year, month, day) {
+    // Since JavaScript is a piece of garbage, months start at 0
```

Comment:
Heretic! Everything should be 0-indexed, not 1-indexed!

Reply:
Why is it not the same for `day` then? :thinking: